bindings/go/oci/internal/lister/component/component_versions.go

```diff
@@ -0,0 +1,82 @@
+package component
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	ociImageSpecV1 "github.com/opencontainers/image-spec/specs-go/v1"
+	"oras.land/oras-go/v2/content"
+	"oras.land/oras-go/v2/registry"
+
+	"ocm.software/open-component-model/bindings/go/oci/internal/lister"
+	"ocm.software/open-component-model/bindings/go/oci/internal/resolver"
+	"ocm.software/open-component-model/bindings/go/oci/spec/annotations"
+	"ocm.software/open-component-model/bindings/go/oci/spec/descriptor"
+)
+
+// ReferrerAnnotationVersionResolver creates a version resolver that extracts component versions
+// from OCI referrer annotations. It validates that the component name matches the expected
+// component and returns the version from the annotation.
+//
+// The annotation format is expected to be: "component-descriptor/<component>:<version>"
+// Returns lister.ErrSkip if the annotation is not present or not in the correct format.
+func ReferrerAnnotationVersionResolver(component string) lister.ReferrerVersionResolver {
+	referrerResolver := func(ctx context.Context, descriptor ociImageSpecV1.Descriptor) (string, error) {
+		if descriptor.Annotations == nil {
+			return "", lister.ErrSkip
+		}
+		annotation, ok := descriptor.Annotations[annotations.OCMComponentVersion]
+		if !ok {
+			return "", lister.ErrSkip
+		}
+		annotation = strings.TrimPrefix(annotation, resolver.DefaultComponentDescriptorPathSuffix+"/")
+		split := strings.Split(annotation, ":")
+		if len(split) != 2 {
+			return "", fmt.Errorf("skipping because not a valid semver tag: %q", annotation)
+		}
```

Comment:
I am confused by the error message. Missing a `:` or having more than one `:` does not necessarily mean that the semver tag is invalid. An invalid semver tag would be something like `v1.002.1` or `asd1.2.asd2`. IMO it means that the annotation has an invalid format.



What do you think?

Reply:
agreed, changed the errors to be more representative